contracts/Rln.sol

```diff
@@ -42,54 +45,48 @@ contract RLN {
 	}
 
 	function _register(uint256 pubkey) internal {
-		members[pubkeyIndex] = pubkey;
+		// Set the pubkey to the value of the tx
+		members[pubkey] = msg.value;
```

Comment:
Because we don't need to store the index. The index is something that the applications require, which, as you mentioned, can be fetched from the tx logs. The model with multiple idCommitments per user which are cycled to create proofs is better than a single idCommitment with different amount staked. 
right now, msg.value will be == the `MEMBERSHIP_DEPOSIT`, however, if we do allow variable registration fee, then we do need to store msg.value :)

This is why I planned to store the msg.value as it may change for different registrations (w/ variable registration fee).

Reply:
Yea, tracking here https://github.com/vacp2p/rln-contract/issues/14